### PR TITLE
Feature: add single head test

### PR DESCRIPTION
**Pull request by caic99**
This PR removes the use of hard-coded `ROOT_PATH`, and takes an abs path to locate the exp runs. The SQL part uses the last level folder name as run_id key, same as before.

### CODE REVIEW

File: src/experiments/run_test.py [hunk 1]
```diff
@@ -1,42 +1,47 @@
...
     temp_file_name = f"{run_id}#{ckpt}#{head}"
     try:
-        process = subprocess.run([ROOT_PATH+"eval_scripts/single_dptest.sh", run_id, str(ckpt), head], check=True) # may need update
-        dptest_res = extract_info_from_dptest_txt(f"{head} ", ROOT_PATH+"eval_scripts/temp_frz_model/"+temp_file_name+".txt") # may need update
-        os.remove(ROOT_PATH+"eval_scripts/temp_frz_model/"+temp_file_name+".txt") # may need update
-        os.remove(ROOT_PATH+"eval_scripts/temp_frz_model/"+temp_file_name+".pth") # may need update
+        script_path = os.path.join(os.path.dirname(__file__), "single_dptest.sh")
+        print(f"Executing command: {script_path} {exp_path} {ckpt} {head}")
+        process = subprocess.run([script_path, exp_path, str(ckpt), head], check=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        outdir="/mnt/workspace/public/multitask/eval_scripts/temp_frz_model/" # from single_dptest.sh
```
Comment (anyangml) [inline src/experiments/run_test.py]: this should be parameterized.

File: src/experiments/run_test.py [hunk 2]
```diff
@@ -1,42 +1,47 @@
...
 import subprocess
 import os
 import numpy as np
-from pathlib import Path
-ROOT_PATH = "/mnt/data_nas/public/multitask/" # may need update
 
 
-def run_single_head_dptest(run_id:str, ckpt:int, head:str):
+def run_single_head_dptest(exp_path:str, ckpt:int, head:str):
     dptest_res = {}
```
Comment (anyangml) [inline src/experiments/run_test.py]: ideally, this should be a general function that can be used for single task test.

File: src/experiments/single_dptest.sh [hunk 3]
```diff
@@ -1,37 +1,31 @@
...
-run_id="$1"
+run_path=$1 # Path storing checkpoints
 ckpt_name=$2
 head_name=$3
-prefix="/mnt/workspace/public/multitask/training_exps/" # may need update
-outdir="/mnt/workspace/public/multitask/eval_scripts/temp_frz_model" # may need update
-run_pth="${prefix}${run_id}"
+outdir="/mnt/workspace/public/multitask/eval_scripts/temp_frz_model/" # may need update
+run_id=$(basename $run_path) # folder name as id
 testfile="/mnt/workspace/public/multitask/eval_scripts/test_files/${head_name}.txt" # may need update
```
Comment (anyangml) [inline src/experiments/single_dptest.sh]: these should be parameterized,  can provide default values in python interface

File: src/experiments/run_test.py (hunk 1 above)
Comment (caic99) [reply src/experiments/run_test.py]: Will do in future

File: src/experiments/run_test.py (hunk 2 above)
Comment (caic99) [reply src/experiments/run_test.py]: Sure! Maybe in another PR?

File: src/experiments/single_dptest.sh (hunk 3 above)
Comment (caic99) [reply src/experiments/single_dptest.sh]: yes, maybe we can pass a `workdir` param through python.

File: src/experiments/run_test.py [hunk 4]
```diff
@@ -1,42 +1,47 @@
...
+            head_dptest_res = run_single_head_dptest(exp_path, ckpt, head)
+
             if np.isnan(head_dptest_res[f"{head}  Virial MAE"]):
                         head_dptest_res[f"{head}  Virial MAE"] = -1
                         head_dptest_res[f"{head}  Virial RMSE"] = -1
                         head_dptest_res[f"{head}  Virial MAE/Natoms"] = -1
                         head_dptest_res[f"{head}  Virial RMSE/Natoms"] = -1
-
+            # print(f"{run_id=}, {head=}, {ckpt=}")
+            # continue # DEBUG
```
Comment (anyangml) [inline src/experiments/run_test.py]: delete

**Comment by caic99** [comment]
Add single test pipeline, tested with `experiments/1018_mptrj_l6_atton_b256_test`

File: lamstare/utils/dptest.py [hunk 5]
```diff
@@ -29,56 +29,40 @@ def extract_info_from_dptest_txt(dataset_name:str, filepath:str="dptest_res.txt"
...
-    """
-    with open(input_file,"r") as f:
-        yaml_dd =  yaml.safe_load(f)
-    mapping = {k:v['filepath'] for k, v in yaml_dd["OOD_TO_HEAD_MAP"].items()}
-    for ood_dataset, filepaths in mapping.items():
-        tem_output_path = f'{output_path}/{ood_dataset}.txt'
-        if Path(tem_output_path).exists() and not overwrite:
-            print(f"File exists at {tem_output_path}. Skipping.")
-            continue
-        print(f"Writing to {tem_output_path}")
```
Comment (anyangml) [inline lamstare/utils/dptest.py]: need to keep this for ood test

**Review by anyangml — APPROVED** [review]

**Pull request merged**